components/camel-google/camel-google-pubsub/src/main/java/org/apache/camel/component/google/pubsub/GooglePubsubConsumer.java

```diff
@@ -177,6 +178,12 @@ private void synchronousPull(String subscriptionName) {
                     }
                 } catch (IOException e) {
                     localLog.error("Failure getting messages from PubSub", e);
+                } catch (ApiException e) {
```

Comment:
Is this exception corresponds to 500 error?

Reply:
@omarsmak Yes, all of the 5xx errors are retryable per https://cloud.google.com/pubsub/docs/reference/error-codes. I initially thought of checking if `e.getStatusCode()` was 5xx, but this seeed better.

Reply:
perfect thanks